resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/KubernetesClientBuilder.scala

```diff
@@ -22,64 +22,62 @@ import com.google.common.base.Charsets
 import com.google.common.io.Files
 import io.fabric8.kubernetes.client.{Config, ConfigBuilder, DefaultKubernetesClient}
 
+import org.apache.spark.SparkConf
+import org.apache.spark.deploy.kubernetes.config._
 import org.apache.spark.deploy.kubernetes.constants._
 
-private[spark] object KubernetesClientBuilder {
+private[spark] class KubernetesClientBuilder(sparkConf: SparkConf, namespace: String) {
   private val SERVICE_ACCOUNT_TOKEN = new File(Config.KUBERNETES_SERVICE_ACCOUNT_TOKEN_PATH)
   private val SERVICE_ACCOUNT_CA_CERT = new File(Config.KUBERNETES_SERVICE_ACCOUNT_CA_CRT_PATH)
-  private val MOUNTED_CREDENTIALS_BASE_DIR = new File(
-    DRIVER_CONTAINER_KUBERNETES_CREDENTIALS_SECRETS_BASE_DIR)
-  private val MOUNTED_TOKEN = new File(MOUNTED_CREDENTIALS_BASE_DIR,
-    DRIVER_CONTAINER_OAUTH_TOKEN_SECRET_NAME)
-  private val MOUNTED_CA_CERT = new File(MOUNTED_CREDENTIALS_BASE_DIR,
-    DRIVER_CONTAINER_CA_CERT_FILE_SECRET_NAME)
-  private val MOUNTED_CLIENT_KEY = new File(MOUNTED_CREDENTIALS_BASE_DIR,
-    DRIVER_CONTAINER_CLIENT_KEY_FILE_SECRET_NAME)
-  private val MOUNTED_CLIENT_CERT = new File(MOUNTED_CREDENTIALS_BASE_DIR,
-    DRIVER_CONTAINER_CLIENT_CERT_FILE_SECRET_NAME)
+  private val oauthTokenFile = sparkConf.get(KUBERNETES_DRIVER_MOUNTED_OAUTH_TOKEN)
+  private val caCertFile = sparkConf.get(KUBERNETES_DRIVER_MOUNTED_CA_CERT_FILE)
+  private val clientKeyFile = sparkConf.get(KUBERNETES_DRIVER_MOUNTED_CLIENT_KEY_FILE)
+  private val clientCertFile = sparkConf.get(KUBERNETES_DRIVER_MOUNTED_CLIENT_CERT_FILE)
 
   /**
-   * Creates a {@link KubernetesClient}, expecting to be from
-   * within the context of a pod. When doing so, credentials files
-   * are picked up from canonical locations, as they are injected
-   * into the pod's disk space.
+   * Creates a {@link KubernetesClient}, expecting to be from within the context of a pod. When
+   * doing so, service account token files can be picked up from canonical locations.
    */
-  def buildFromWithinPod(kubernetesNamespace: String): DefaultKubernetesClient = {
-    var clientConfigBuilder = new ConfigBuilder()
+  def buildFromWithinPod(): DefaultKubernetesClient = {
+    val baseClientConfigBuilder = new ConfigBuilder()
       .withApiVersion("v1")
       .withMasterUrl(KUBERNETES_MASTER_INTERNAL_URL)
-      .withNamespace(kubernetesNamespace)
+      .withNamespace(namespace)
 
-    if (MOUNTED_TOKEN.isFile ||
-        MOUNTED_CA_CERT.isFile ||
-        MOUNTED_CLIENT_KEY.isFile ||
-        MOUNTED_CLIENT_CERT.isFile) {
-      if (MOUNTED_TOKEN.isFile) {
-        clientConfigBuilder = clientConfigBuilder.withOauthToken(
-          Files.toString(MOUNTED_TOKEN, Charsets.UTF_8))
+    val configBuilder = oauthTokenFile
+        .orElse(caCertFile)
+        .orElse(clientKeyFile)
+        .orElse(clientCertFile)
```

Comment:
Looks like the Scala way (from 2011 via http://stackoverflow.com/questions/6390797/scala-any-and-all-functions and 2013 via http://stackoverflow.com/questions/15932137/checking-that-all-items-in-a-collection-match-a-predicate-in-scala) is:

`Seq(oauthTokenFile, caCertFile, clientKeyFile, clientCertFile).any(_.isDefined)`

Do you find that clearer than the .orElse chain ?